content/hi/docs/tasks/extend-kubectl/kubectl-plugins.md

```diff
@@ -0,0 +1,367 @@
+इस कमांड को एक्जीक्यूट करने से आपके `PATH` में सभी फाइलों का ट्रैवर्सल होता है। कोई भी फाइल जो एक्जीक्यूटेबल है और
+`kubectl-` से शुरू होती है, वह इस कमांड के आउटपुट में *आपके `PATH` में मौजूद क्रम में* दिखाई देगी।
+`kubectl-` से शुरू होने वाली किसी भी फाइल के लिए एक चेतावनी शामिल की जाएगी जो *एक्जीक्यूटेबल नहीं* है।
+ऐसी किसी भी वैध प्लगइन फाइल के लिए भी एक चेतावनी शामिल की जाएगी जो एक-दूसरे के नाम को ओवरलैप करती हैं।
+
+आप [Krew](https://krew.dev/) का उपयोग कम्युनिटी द्वारा क्यूरेट किए गए [plugin index](https://krew.sigs.k8s.io/plugins/) से `kubectl` प्लगइन्स को खोजने और इंस्टॉल करने के लिए कर सकते हैं।
+
+#### प्लगइन्स बनाना
+
+`kubectl` प्लगइन्स को `PATH` में `kubectl-create-something` बाइनरी प्रदान करके `kubectl create something` के आकार में कस्टम क्रिएट कमांड्स जोड़ने की अनुमति देता है।
+
+#### सीमाएं
+
+वर्तमान में मौजूदा `kubectl` कमांड्स को ओवरराइट करने या `create` के अलावा अन्य कमांड्स को विस्तारित करने वाले प्लगइन बनाना संभव नहीं है।
+उदाहरण के लिए, एक प्लगइन `kubectl-version` बनाने से वह प्लगइन कभी भी एक्जीक्यूट नहीं होगा, क्योंकि मौजूदा `kubectl version`
+कमांड हमेशा इसकी तुलना में प्राथमिकता लेगा।
+इस सीमा के कारण, मौजूदा `kubectl` कमांड्स में नए सबकमांड जोड़ने के लिए प्लगइन्स का उपयोग करना भी *संभव नहीं* है।
+उदाहरण के लिए, अपने प्लगइन का नाम `kubectl-attach-vm` रखकर एक सबकमांड `kubectl attach vm` जोड़ने से वह प्लगइन नजरअंदाज कर दिया जाएगा।
+
+`kubectl plugin list` ऐसा करने का प्रयास करने वाले किसी भी वैध प्लगइन के लिए चेतावनियां दिखाता है।
+
+## kubectl प्लगइन्स लिखना
+
+आप किसी भी प्रोग्रामिंग भाषा या स्क्रिप्ट में प्लगइन लिख सकते हैं जो आपको कमांड-लाइन कमांड्स लिखने की अनुमति देती है।
+
+कोई प्लगइन इंस्टॉलेशन या पूर्व-लोडिंग आवश्यक नहीं है। प्लगइन एक्जीक्यूटेबल्स को `kubectl` बाइनरी से विरासत में मिला पर्यावरण प्राप्त होता है।
+एक प्लगइन अपने नाम के आधार पर तय करता है कि वह किस कमांड पाथ को लागू करना चाहता है।
+उदाहरण के लिए, `kubectl-foo` नाम का एक प्लगइन `kubectl foo` कमांड प्रदान करता है। आपको
+प्लगइन एक्जीक्यूटेबल को अपने `PATH` में कहीं भी इंस्टॉल करना होगा।
+
+### उदाहरण प्लगइन
+
+```bash
+#!/bin/bash
+
+# वैकल्पिक आर्गुमेंट हैंडलिंग
+if [[ "$1" == "version" ]]
+then
+    echo "1.0.0"
+    exit 0
+fi
+
+# वैकल्पिक आर्गुमेंट हैंडलिंग
+if [[ "$1" == "config" ]]
+then
+    echo "$KUBECONFIG"
+    exit 0
+fi
+
+echo "मैं kubectl-foo नाम का एक प्लगइन हूं"
+```
+
+### प्लगइन का उपयोग करना
+
+प्लगइन का उपयोग करने के लिए, प्लगइन को एक्जीक्यूटेबल बनाएं:
+
+```shell
+sudo chmod +x ./kubectl-foo
+```
+
+और इसे अपने `PATH` में कहीं भी रखें:
+
+```shell
+sudo mv ./kubectl-foo /usr/local/bin
+```
+
+अब आप अपने प्लगइन को एक `kubectl` कमांड के रूप में इनवोक कर सकते हैं:
+
+```shell
+kubectl foo
+```
+
+```
+मैं kubectl-foo नाम का एक प्लगइन हूं
+```
+
+सभी आर्ग्स और फ्लैग्स जैसे के तैसे एक्जीक्यूटेबल को पास किए जाते हैं:
+
+```shell
+kubectl foo version
+```
+
+```
+1.0.0
+```
+
+सभी एनवायरनमेंट वेरिएबल्स भी जैसे के तैसे एक्जीक्यूटेबल को पास किए जाते हैं:
```

Comment:
Here you've mentioned environment, above in line #61 you've translated to पर्यावरण. Let's maintain one or another, please.